backend/src/types/index.js

```diff
@@ -0,0 +1,30 @@
+import fs from 'fs'
+import path from 'path'
+import { mergeTypes } from 'merge-graphql-schemas'
+
+const findGqlFiles = dir => {
+  var results = []
+  var list = fs.readdirSync(dir)
+  list.forEach(file => {
+    file = path.join(dir, file).toString('utf-8')
+    var stat = fs.statSync(file)
+    if (stat && stat.isDirectory()) {
+      // Recurse into a subdirectory
+      results = results.concat(findGqlFiles(file))
+    } else {
+      if (path.extname(file) === '.gql') {
+        // Is a gql file
+        results.push(file)
+      }
+    }
+  })
+  return results
+}
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-31T11:50:25Z" title="Friday, May 31st 2019, 1:50:25 pm +02:00">May 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,30 @@
+import fs from 'fs'
+import path from 'path'
+import { mergeTypes } from 'merge-graphql-schemas'
+
+const findGqlFiles = dir => {
+  var results = []
+  var list = fs.readdirSync(dir)
+  list.forEach(file => {
+    file = path.join(dir, file).toString('utf-8')
+    var stat = fs.statSync(file)
+    if (stat && stat.isDirectory()) {
+      // Recurse into a subdirectory
+      results = results.concat(findGqlFiles(file))
+    } else {
+      if (file.split('.').pop() === 'gql') {
+        // Is a gql file
+        results.push(file)
+      }
+    }
+  })
+  return results
+}
```


@ulfgebhardt great! Can we do sth. similar [here](https://github.com/Human-Connection/Human-Connection/blob/master/backend/src/graphql-schema.js#L41) ?

Reply:
> [<img alt="ulfgebhardt" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-31T14:06:21Z" title="Friday, May 31st 2019, 4:06:21 pm +02:00">May 31, 2019</time>_

---


yes